UnitsNet/CustomCode/Quantity.cs

```diff
@@ -199,38 +187,30 @@ public static bool TryFromUnitAbbreviation(IFormatProvider? formatProvider, doub
         /// <exception cref="UnitNotFoundException">Type must be of type UnitsNet.IQuantity -or- Type is not a known quantity type.</exception>
         public static IQuantity Parse(IFormatProvider? formatProvider, Type quantityType, string quantityString)
         {
-            return Default.Parse(formatProvider, quantityType, quantityString);
+            // TODO Support custom units (via the QuantityParser), currently only hardcoded built-in quantities are supported.
+            if (!typeof(IQuantity).IsAssignableFrom(quantityType))
+                throw new ArgumentException($"Type {quantityType} must be of type UnitsNet.IQuantity.");
+
+            if (TryParse(formatProvider, quantityType, quantityString, out IQuantity? quantity))
+                return quantity;
+
+            throw new UnitNotFoundException($"Quantity string '{quantityString}' could not be parsed to quantity '{quantityType}'.");
```

Comment:
In my other project- the `QuantityParser` can directly instantiate an `IQuantity` (by calling the `QuantityInfoLookup.From(..)`).